src/platform/windows/mod.rs

```diff
@@ -126,6 +131,37 @@ pub fn channel() -> Result<(OsIpcSender, OsIpcReceiver), WinError> {
     Ok((sender, receiver))
 }
 
+/// Unify the creation of sender and receiver duplex pipes to allow for either to be spawned first.
+/// Requires the use of a duplex and therefore lets both sides read and write.
+unsafe fn create_duplex(pipe_name: &CString) -> Result<HANDLE, WinError> {
+    if let Ok(handle) = CreateFileA(
+        PCSTR::from_raw(pipe_name.as_ptr() as *const u8),
+        FILE_GENERIC_WRITE.0 | FILE_GENERIC_READ.0,
+        FILE_SHARE_MODE(0),
+        None, // lpSecurityAttributes
+        OPEN_EXISTING,
+        FILE_ATTRIBUTE_NORMAL,
+        None,
+    ) {
+        Ok(handle)
+    } else {
+        let handle = CreateNamedPipeA(
+            PCSTR::from_raw(pipe_name.as_ptr() as *const u8),
+            PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
+            PIPE_TYPE_BYTE | PIPE_READMODE_BYTE | PIPE_REJECT_REMOTE_CLIENTS,
+            // 1 max instance of this pipe
+            1,
+            // out/in buffer sizes
+            0,
+            PIPE_BUFFER_SIZE as u32,
+            0, // default timeout for WaitNamedPipe (0 == 50ms as default)
+            None,
+        )?;
+
+        Ok(handle)
+    }
```

Comment:
This can use `or`:

```suggestion
    CreateFileA(
        PCSTR::from_raw(pipe_name.as_ptr() as *const u8),
        FILE_GENERIC_WRITE.0 | FILE_GENERIC_READ.0,
        FILE_SHARE_MODE(0),
        None, // lpSecurityAttributes
        OPEN_EXISTING,
        FILE_ATTRIBUTE_NORMAL,
        None,
    )
    .or(CreateNamedPipeA(
        PCSTR::from_raw(pipe_name.as_ptr() as *const u8),
        PIPE_ACCESS_DUPLEX | FILE_FLAG_OVERLAPPED,
        PIPE_TYPE_BYTE | PIPE_READMODE_BYTE | PIPE_REJECT_REMOTE_CLIENTS,
        // 1 max instance of this pipe
        1,
        // out/in buffer sizes
        0,
        PIPE_BUFFER_SIZE as u32,
        0, // default timeout for WaitNamedPipe (0 == 50ms as default)
        None,
    ))
```

Reply:
Of course, is there any advantage between the two besides readability?  Ill just commit it, but I am curious.

Reply:
I will change it to use 'or_else' in a follow up PR.  We certainly don't want it to be eagerly evaluated.

Reply:
Fixed in #391.